lib/dao.js

```diff
@@ -211,32 +211,36 @@ DataAccessObject.create = function (data, callback) {
       callback(new ValidationError(obj), obj);
     }
   }, data);
-  
+
   function create() {
     obj.trigger('create', function (createDone) {
       obj.trigger('save', function (saveDone) {
-
-        var _idName = idName(Model);
-        var modelName = Model.modelName;
-        this._adapter().create(modelName, this.constructor._forDB(obj.toObject(true)), function (err, id, rev) {
-          if (id) {
-            obj.__data[_idName] = id;
-            defineReadonlyProp(obj, _idName, id);
-          }
-          if (rev) {
-            obj._rev = rev;
-          }
-          if (err) {
-            return callback(err, obj);
-          }
-          obj.__persisted = true;
-          saveDone.call(obj, function () {
-            createDone.call(obj, function () {
-              callback(err, obj);
-              if(!err) Model.emit('changed', obj);
+        var self = this;
+        Model.notify('before save', { model: obj }, function(err) {
```

Comment:
`before save` ATM. I am open to other conventions like `before-save`, `save:before` or even `saving` (before) and `saved` (after). I'd like to avoid `beforeSave` as it may be confused with the old `Model.beforeSave` hook.